File: src/06/3/parameters.php
<?php
$firstName = '';
$lastName = '';
if (isset($_GET['firstname'])) $firstName = $_GET['firstname'];
if (isset($_GET['lastname'])) $lastName = $_GET['lastname'];
?>
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>リクエスト内容の取得</title>
  </head>
  <body>
    <?php
    echo '<dl>';
    echo '<dt>First Name</dt>';
    echo '<dd>' . htmlspecialchars($firstName, ENT_QUOTES, 'UTF-8') . '</dd>';
    echo '<dt>Last Name</dt>';
    echo '<dd>' . htmlspecialchars($lastName, ENT_QUOTES, 'UTF-8') . '</dd>';
    echo '</dl>';
    ?>
  </body>
</html>

File: src/06/3/Parameters.java
import java.io.*;
import javax.servlet.*;
import javax.servlet.http.*;
import javax.servlet.annotation.*;
import org.apache.commons.lang.*;
 
@WebServlet(name = "Parameters", urlPatterns = {"/Parameters"})
public class Parameters extends HttpServlet {
 
  @Override
  protected void doGet(HttpServletRequest request, HttpServletResponse response)
          throws ServletException, IOException {
    request.setCharacterEncoding("UTF-8");
    String firstName = request.getParameter("firstname");
    String lastName = request.getParameter("lastname");
 
    PrintWriter out = response.getWriter();
    out.println("<html><body><dl>");
    out.println("<dt>First Name</dt>");
    out.println("<dd>" + StringEscapeUtils.escapeXml(firstName) + "</dd>");
    out.println("<dt>Last Name</dt>");
    out.println("<dd>" + StringEscapeUtils.escapeXml(lastName) + "</dd>");
    out.println("</dl></body></html>");
  }
}

